data/devices/data-parse-test.py

```diff
@@ -190,6 +191,16 @@ def check_section_driver(driver, section):
         assert('Unsupported driver section {}'.format(driver))
 
 
+def validate_data_file_name(path):
+    # Matching any of the characters in the regular expression will throw an
+    # error. Currently only tests the square brackets [], parentheses, and curly
+    # braces.
+    illegal_characters_regex = '([\[\]\{\}\(\)])'
+    found_characters = re.findall(illegal_characters_regex, path)
+    if len(found_characters) > 0:
+        raise AssertionError('data file name \'{}\' contains illegal characters: \'{}\''.format(path, ''.join(found_characters)))
```

Comment:
I would use `if found_characters` and double quotes in the `AssertionError` to avoid having to escape the single quotes.

Reply:
No sense in having me muck up something that would be cleaner otherwise. One more run on this guy then.